From d55e35cf53e47e55bca4f709448f9a282964ddc4 Mon Sep 17 00:00:00 2001
From: Pranav Tipnis <pranav.tipnis@intel.com>
Date: Thu, 1 Dec 2016 10:19:21 -0800
Subject: [PATCH 27/27] acpi: Workaround for not registering CAN controller

This patch skips adding ACPI device for MCHP2515 CAN
controller. CAN controller will be accessed via spidev
interface instead.

Signed-off-by: Pranav Tipnis <pranav.tipnis@intel.com>
---
 drivers/acpi/scan.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/acpi/scan.c b/drivers/acpi/scan.c
index e57bd86..9fb5a48 100644
--- a/drivers/acpi/scan.c
+++ b/drivers/acpi/scan.c
@@ -611,6 +611,13 @@ int acpi_device_add(struct acpi_device *device,
 	struct acpi_device_bus_id *acpi_device_bus_id, *new_bus_id;
 	int found = 0;
 
+	/* Workaround: Do not add MCHP2515 CAN controller */
+	struct acpi_hardware_id *hwid;
+	list_for_each_entry(hwid, &device->pnp.ids, list)
+		if (!strcmp("MCHP2515", hwid->id))
+			return -ENODEV;
+	/* End of Workaround */
+
 	if (device->handle) {
 		acpi_status status;
 
-- 
2.9.3

